core/core.go

```diff
@@ -210,6 +248,46 @@ func (n *IpfsNode) OnlineMode() bool {
 	return n.onlineMode
 }
 
+func (n *IpfsNode) Bootstrap(ctx context.Context, peers []peer.PeerInfo) error {
+	if n.DHT != nil {
+		for _, p := range peers {
+			// TODO bootstrap(ctx, n.PeerHost, n.DHT, n.Peerstore, peers)
+			if err := n.DHT.Connect(ctx, p.ID); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// TODO we may not want to add these methods to the core. Maybe they should be
+// defined as free functions in another package that use public fields on the
+// node.
+//
+// e.g. reader, err := unix.Cat(node)
+
+func (n *IpfsNode) Cat(k u.Key) (io.Reader, error) {
+	catterdag := n.DAG
+	nodeCatted, err := (&path.Resolver{catterdag}).ResolvePath(k.String())
+	if err != nil {
+		return nil, err
+	}
+	return uio.NewDagReader(nodeCatted, catterdag)
+}
```

Comment:
this is beautiful. and yes, +1 on free functions. btw, lots of it can probably be abstracted into an interface (i.e. `cat` shouldnt need to know what a `path.Resolver` is, and instead could call `ipfs.Resolve(path)`)